modules/agent-policy/scripts/script-utils.sh

```diff
@@ -0,0 +1,339 @@
+#   $1 = flag name
+#   $2 = flag value
+# Return:
+#   An appropriate --clear-x flag (where x is instances, group-labels, or zones)
+#   if the flag value is empty, otherwise returns the appropriate flag
+function get_update_flag() {
+    # if the flag is empty
+    # we return the "clear group labels flag"
+    local flag_name="$1"
+    local flag_value="$2"
+    local update_flag=""
+    if [ -z "$flag_value" ]; then
+        # flag value is empty
+        update_flag=" --clear-$flag_name"
+    fi
+    echo "$update_flag"
+}
+
+
+# Params:
+#   $1 = group labels flag name
+#   $2 = group labels flag value
+#   $3 = zones flag name
+#   $4 = zones flag value
+#   $5 = instances flag name
+#   $5 = instances flag value
+# Return:
+#   The appropriate --clear-x flags (where x is group-labels, zones, or instances)
+#   based on grloup labels flag value, zones flag value, and instances flag value
+function get_update_flags() {
+    local group_labels_flag_name="$1"
+    local group_labels_flag_value="$2"
+    local zones_flag_name="$3"
+    local zones_flag_value="$4"
+    local instances_flag_name="$5"
+    local instances_flag_value="$6"
+    local clear_group_labels_flag
+    local clear_zones_flag
+    local update_flags
+
+    clear_group_labels_flag="$(get_update_flag "$group_labels_flag_name" \
+        "$group_labels_flag_value")"
+    clear_zones_flag=$(get_update_flag "$zones_flag_name" "$zones_flag_value")
+    clear_instances_flag=$(get_update_flag "$instances_flag_name" "$instances_flag_value")
+
+    local update_flags="$clear_group_labels_flag$clear_zones_flag$clear_instances_flag"
+    echo "$update_flags"
+}
+
+# Params:
+#   $1 = project id
+# Return:
+#   The appropriate global flags (--project and --quiet)
+function get_global_flags() {
+    local project_id="$1"
+    local project_flag_name="project"
+    local project_flag
+    project_flag=$(get_flag "$project_flag_name" "$project_id")
+    local quiet_flag=" --quiet"
+
+    local global_flags="$project_flag$quiet_flag"
+    echo "$global_flags"
+}
+
+# Params:
+#   $1 = action (create or update)
+#   $2 = policy id
+#   $3 = description of the agent policy
+#   $4 = agent rules, in json format
+#   $5 = group labels, in json format
+#   $6 = os types, in json format
+#   $7 = zones, in json format
+#   $8 = instances, in json format
+# Return:
+#   the appropriate gcloud create or update command, given the args
+function get_base_upsert_command() {
+    local action="$1"
+    local policy_id="$2"
+    local description="$3"
+    local agent_rules_json="$4"
+    local group_labels_json="$5"
+    local os_types_json="$6"
+    local zones_json="$7"
+    local instances_json="$8"
+
+    local description_flag_name="description"
+    local agent_rules_flag_name="agent-rules"
+    local group_labels_flag_name="group-labels"
+    local os_types_flag_name="os-types"
+    local zones_flag_name="zones"
+    local instances_flag_name="instances"
+
+    local agent_rules_flag_value
```

Comment:
What is the benefit of declaring all of these beforehand, rather than simply initializing them in a single line? Similar to how you do the flag_name variables above.

Reply:
Yeah, it's to avoid masking return values. The linter checks for this as well (this link also explains why masking return values is bad): https://github.com/koalaman/shellcheck/wiki/SC2155